File: lib/helper/transaction.dart
class Transaction {
  String id;
  String title;
  double amount;
  String type; // income or expense
  String categoryId;
  String categoryName;
  DateTime date;

  Transaction({
    required this.id,
    required this.title,
    required this.amount,
    required this.type,
    required this.categoryId,
    required this.categoryName,
    required this.date,
  });

  factory Transaction.fromJson(Map<String, dynamic> json) {
    return Transaction(
      id: json['_id'] ?? '',
      title: json['title'] ?? '',
      amount: (json['amount'] ?? 0).toDouble(),
      type: json['type'] ?? 'expense',
      categoryId: json['categoryId'] ?? '',
      categoryName: json['categoryName'] ?? '',
      date: json['date'] != null 
          ? DateTime.parse(json['date']) 
          : DateTime.now(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'title': title,
      'amount': amount,
      'type': type,
      'categoryId': categoryId,
      'categoryName': categoryName,
      'date': date.toIso8601String(),
    };
  }
}

File: server/models/Transaction.js
const mongoose =  require("mongoose")

const transactionSchema = new mongoose.Schema({

    title:{
        type: String,
        required: true,
    },
    amount:{
     type:  Number,
     required: true,
    },
    type: {
type: String,
enum: ["income","expense"], required: true
    },

    category: {
        type : mongoose.Schema.Types.ObjectId,
        ref : "Category"
    },
    date: {
        type: Date,
        default: Date.now
    }



})


const Transaction = mongoose.model("Transaction", transactionSchema);
module.exports = Transaction;